include/NovelRT/Graphics/Camera.h

```diff
@@ -0,0 +1,61 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root for more information.
+
+#ifndef NOVELRT_GRAPHICS_CAMERA_H
+#define NOVELRT_GRAPHICS_CAMERA_H
+#ifndef NOVELRT_H
+#error Please do not include this directly. Use the centralised header (NovelRT.h) instead!
+#endif
+
+namespace NovelRT::Graphics {
+  class Camera {
+
+    NOVELRT_EVENT(CameraViewChanged, CameraViewChangedEventArgs)
+
+  //TODO: Look Im just trying to make this legal code okay?
+  public:
+    enum class CameraMode : unsigned int {
+      None,
+      Orthographic,
+      Perspective
+    };
+
+  private:
+
+    Maths::GeoMatrix4<float> _viewMatrix;
+    Maths::GeoMatrix4<float> _projectionMatrix;
+    Utilities::Lazy<Maths::GeoMatrix4<float>> _cameraUboMatrix;
+
+    Maths::GeoMatrix4<float> generateUboMatrix();
+    CameraMode _cameraMode;
+
+  public:
+
+    Camera();
+
+    inline Maths::GeoMatrix4<float> getViewMatrix() const {
+      return _viewMatrix;
+    }
+
+    //TODO: Jeez this is written badly
+    inline void setViewMatrix(Maths::GeoMatrix4<float> value) {
+      _cameraUboMatrix.reset();
+      _viewMatrix = value;
+      raiseCameraViewChanged(CameraViewChangedEventArgs(getCameraUboMatrix()));
+    }
+
+    inline Maths::GeoMatrix4<float> getProjectionMatrix() const {
+      return _projectionMatrix;
+    }
+
+    inline void setProjectionMatrix(Maths::GeoMatrix4<float> value) {
+      _cameraUboMatrix.reset();
+      _projectionMatrix = value;
+      raiseCameraViewChanged(CameraViewChangedEventArgs(getCameraUboMatrix()));
+    }
+
+    inline Maths::GeoMatrix4<float> getCameraUboMatrix() { //TODO: I'm unsure about getters like these since its coming from a Lazy. Thoughts?
```

Comment:
you mean since it's not a ref return?

Reply:
Correct. This is returning from the lazy but do we forward that by reference?

Reply:
We could have the lazy expose a `T&` (and still ensure it is a valid one by enforcing initialization).

Reply:
Just checked the header - seems `getActual` is already returning a `T&`. I'll just forward it then. Awesome. 😄